cli/tests/pcluster/templates/test_cluster_stack.py

```diff
@@ -24,3 +28,126 @@ def test_cluster_builder(mocker):
+    ],
+)
+def test_pcluster_dashboard_builder(test_datadir, config_file_name):
+    mock_aws_api()
+    input_yaml = load_yaml_dict(test_datadir / config_file_name)
+    cluster_config = ClusterSchema().load(input_yaml)
+    print(cluster_config)
+    generated_template = CDKTemplateBuilder().build_cluster_template(
+        cluster_config=cluster_config, bucket=dummy_bucket(), stack_name="parallelcluster-dummyname"
+    )
+    output_yaml = yaml.dump(generated_template, width=float("inf"))
+    print(output_yaml)
+
+    if condition_cw_dashboard_enabled(cluster_config):
+        if cluster_config.shared_storage:
+            _verify_ec2_metrics_conditions(cluster_config, output_yaml)
+
+        if condition_cw_logging_enabled(cluster_config):
+            _verify_head_node_logs_conditions(cluster_config, output_yaml)
+        else:
+            assert_that(cluster_config).does_not_contain("Head Node Logs")
+
+
+def _verify_ec2_metrics_conditions(cluster_config, output_yaml):
+    storage_resource = {storage_type: [] for storage_type in SharedStorageType}
+    storage_type_title_dict = {
+        SharedStorageType.EBS: "EBS Metrics",
+        SharedStorageType.RAID: "RAID Metrics",
+        SharedStorageType.EFS: "EFS Metrics",
+        SharedStorageType.FSX: "FSx Metrics",
+    }
+
+    for storage in cluster_config.shared_storage:
+        storage_resource[storage.shared_storage_type].append(storage)
+
+    # Check each section title
+    for storage_type, storages in storage_resource.items():
+        if len(storages) > 0:
+            assert_that(output_yaml).contains(storage_type_title_dict[storage_type])
+        else:
+            assert_that(output_yaml).does_not_contain(storage_type_title_dict[storage_type])
+
+    # Conditional EBS and RAID metrics
+    ebs_and_raid_storage = storage_resource[SharedStorageType.EBS] + storage_resource[SharedStorageType.RAID]
+    if any(storage.volume_type == "io1" for storage in ebs_and_raid_storage):
+        assert_that(output_yaml).contains("Consumed Read/Write Ops")
+        assert_that(output_yaml).contains("Throughput Percentage")
+    else:
+        assert_that(output_yaml).does_not_contain("Consumed Read/Write Ops")
+        assert_that(output_yaml).does_not_contain("Throughput Percentage")
+
+    burst_balance = any(storage.volume_type in ["gp1", "st1", "sc1"] for storage in ebs_and_raid_storage)
+    if burst_balance:
+        assert_that(output_yaml).contains("Burst Balance")
+    else:
+        assert_that(output_yaml).does_not_contain("Burst Balance")
+
+    # conditional EFS metrics
+    percent_io_limit = any(
+        storage.performance_mode == "generalPurpose" for storage in storage_resource[SharedStorageType.EFS]
+    )
+    if percent_io_limit:
+        assert_that(output_yaml).contains("PercentIOLimit")
+    else:
+        assert_that(output_yaml).does_not_contain("PercentIOLimit")
+
+
+def _verify_head_node_logs_conditions(cluster_config, output_yaml):
```

Comment:
You can delete the old test files in the cloudformation tests given that you migrated them.

Reply:
Done